### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by saidou224**
to check 

- code quality 
- and structure of classes

### CODE REVIEW

File: src/main/java/com/sba/hockeyGame/domain/model/Team.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  @Column(name = "id", nullable = false)
+  private Long id;
+
+  @Column(name = "coach", nullable = false)
+  private String coach;
+
+  @Column(name = "year", nullable = false, unique = true)
+  private Long year;
+
+  @OneToMany(fetch = FetchType.EAGER, mappedBy = "team", cascade = CascadeType.ALL)
```
Comment (hjalloh) [inline src/main/java/com/sba/hockeyGame/domain/model/Team.java:38]: Do you always wanna load the players when loading a team?

File: src/main/java/com/sba/hockeyGame/domain/services/impl/PlayerServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  private final PlayerMapper playerMapper;
+  private final PlayerQuery playerQuery;
+
+
+  @Override
+  public Player makePlayerCaptain(Long playerId) throws InstanceAlreadyExistsException {
+
+
+    Player player = playerQuery.findById(playerId)
+            .orElseThrow(() -> new NoSuchElementException(
```
Comment (hjalloh) [inline src/main/java/com/sba/hockeyGame/domain/services/impl/PlayerServiceImpl.java:35]: Are you handling this exception in your exception handler?

File: src/main/java/com/sba/hockeyGame/domain/services/impl/TeamServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  @Override
+  public List<Team> getTeams(int pageNumber, int pageSize) {
+    return teamQuery.retrieveTeams(pageNumber, pageSize).getContent();
+  }
+
+  @Override
+  public Team findTeamByYear(Long year) {
+    return teamQuery.findTeamByYear(year)
+            .orElseThrow(() -> new NoSuchElementException(
```
Comment (hjalloh) [inline src/main/java/com/sba/hockeyGame/domain/services/impl/TeamServiceImpl.java:39]: Same, is this exception caught by the exception handler?

File: src/main/java/com/sba/hockeyGame/domain/utils/ApiUtils.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.List;
+import javax.management.InstanceAlreadyExistsException;
+
+public class ApiUtils {
+
+  public static void checkTeamCaptainAlreadyExist(final Long yearOfTeam, final boolean isCaptain, final List<Player> players)
+      throws InstanceAlreadyExistsException {
+    if (isCaptain &&
+        players.stream()
+            .anyMatch(player -> Boolean.TRUE == player.getIsCaptain())) {
```
Comment (hjalloh) [inline src/main/java/com/sba/hockeyGame/domain/utils/ApiUtils.java:13]: the lambda expression cannot be simplified by method reference like 
```suggestion
            .anyMatch(player::getIsCaptain)) {
```

**Comment by hjalloh** [comment]
your package structure looks a bit inconsistent:

domain/<DOMAIN-TYPE>/<Business-Oriented-Class> (eg: domain/model/{Player, Team, etc}
    VS
domain/<Business-Type>/<Feature-Oriented-Class> (eg: domain/player/{PlayerCommand, PlayerRepository, etc}


image

File: src/main/java/com/sba/hockeyGame/domain/model/Team.java (hunk 1 above)
Comment (saidou224) [reply src/main/java/com/sba/hockeyGame/domain/model/Team.java:38]: yes

File: src/main/java/com/sba/hockeyGame/domain/services/impl/PlayerServiceImpl.java (hunk 2 above)
Comment (saidou224) [reply src/main/java/com/sba/hockeyGame/domain/services/impl/PlayerServiceImpl.java:35]: yes

File: src/main/java/com/sba/hockeyGame/domain/services/impl/TeamServiceImpl.java (hunk 3 above)
Comment (saidou224) [reply src/main/java/com/sba/hockeyGame/domain/services/impl/TeamServiceImpl.java:39]: yes

File: src/main/java/com/sba/hockeyGame/domain/utils/ApiUtils.java (hunk 4 above)
Comment (saidou224) [reply src/main/java/com/sba/hockeyGame/domain/utils/ApiUtils.java]: good advice

**Review by hjalloh — APPROVED** [review]

**Pull request merged**